core/services/job/spawner.go

```diff
@@ -125,11 +128,28 @@ func (js *spawner) runLoop() {
 		defer newJobs.Close()
 		newJobEvents = newJobs.Events()
 	}
+	var pgDeletedJobEvents <-chan postgres.Event
+	deletedJobs, err := js.orm.ListenForDeletedJobs()
+	if err != nil {
+		logger.Warn("Job spawner could not subscribe to deleted job events")
+	} else {
+		defer deletedJobs.Close()
+		pgDeletedJobEvents = deletedJobs.Events()
+	}
 
 	// Initialize the DB poll ticker
 	dbPollTicker := time.NewTicker(js.config.JobPipelineDBPollInterval())
 	defer dbPollTicker.Stop()
 
+	// Initialize the poll that checks for deleted jobs and removes them
+	// This is only necessary as a fallback in case the event doesn't fire for some reason
```

Comment:
One such reason would be if the node happens to be down while the delete happens, then comes back up after right?

Reply:
Yup

Reply:
worth a comment

Reply:
Not quite, that scenario will be handled automatically because the removed job simply will not be loaded on restart.

This poll is in case the trigger fails (e.g. temporarily lost connection to database). This is likely exceedingly rare but its good to have backups in the any subscription service with best effort delivery.